### PR TITLE
sygus: Refactor to not use NodeManager::currentNM()

### PR DESCRIPTION
**Pull request by daniel-larraz**

### CODE REVIEW

File: src/theory/quantifiers/sygus/cegis_core_connective.cpp [hunk 1]
```diff
@@ -108,7 +110,7 @@ bool CegisCoreConnective::processInitialize(Node conj,
   std::vector<Node> echildren;
   echildren.push_back(d_candidate);
   echildren.insert(echildren.end(), d_vars.begin(), d_vars.end());
-  d_eterm = NodeManager::currentNM()->mkNode(Kind::DT_SYGUS_EVAL, echildren);
+  d_eterm = nodeManager()->mkNode(Kind::DT_SYGUS_EVAL, echildren);
```
Comment (aniemetz) [inline src/theory/quantifiers/sygus/cegis_core_connective.cpp:113]: This class has a node manager as a member, right? All its member functions should use `d_nm` directly.
```suggestion
  d_eterm = d_nm->mkNode(Kind::DT_SYGUS_EVAL, echildren);
```

File: src/theory/quantifiers/sygus/cegis_core_connective.cpp [hunk 2]
```diff
@@ -379,7 +381,7 @@ Node CegisCoreConnective::Component::getSygusSolution(
   std::sort(conjs.begin(), conjs.end());
   Node sol;
   std::map<Node, Node>::const_iterator itu;
-  NodeManager* nm = NodeManager::currentNM();
+  NodeManager* nm = d_nm;
```
Comment (aniemetz) [inline src/theory/quantifiers/sygus/cegis_core_connective.cpp:384]: Same as above, why not use `d_nm` directly?

**Review by aniemetz — CHANGES_REQUESTED** [review]

File: src/theory/quantifiers/sygus/cegis_core_connective.cpp (hunk 1 above)
Comment (daniel-larraz) [reply src/theory/quantifiers/sygus/cegis_core_connective.cpp:113]: No, the `CegisCoreConnective` class does not have a node manager as a direct member (`d_nm`). Instead, the class inherits from `EnvObj` via `Cegis` and `SygusModule`. The `EnvObj` class includes an associated environment, `d_env`, as a member. Therefore, you can either call `nodeManager()` directly or access it indirectly via `d_env.getNodeManager()`. However, the latter approach might be less desirable.

File: src/theory/quantifiers/sygus/cegis_core_connective.cpp (hunk 2 above)
Comment (daniel-larraz) [reply src/theory/quantifiers/sygus/cegis_core_connective.cpp]: This case is different. `CegisCoreConnective::Component` has a `d_nm` member. Changed.

Comment (daniel-larraz) [reply src/theory/quantifiers/sygus/cegis_core_connective.cpp]: After merging PR #11403 , the member `d_nm` is no longer necessary.

**Review by aniemetz — APPROVED** [review]

**Pull request merged**